entry.c

```diff
@@ -366,6 +366,9 @@ static int write_entry(struct cache_entry *ce,
 	}
 
 finish:
+	// Flush cached lstat in fscache after writing disk.
+	flush_fscache();
```

Comment:
So this tries to flush the entire FS Cache every single time a file is written from an index entry. I wonder whether that is a bit heavy-handed, in particular since we will hit this code path in `git reset` (where the FS Cache is currently *not* used, yet the `fscache_clear()` would implicitly initialize the fscache hashmap).

I had hoped that we could somehow update the cached data after writing the file, but I guess we cannot really do this elegantly?

Reply:
For `git reset`, `flush_fscache` does not call `fscache_clear()` when fscache is disabled.
I think it does not become large overhead.

Incremental update is difficult for me without large modification of current fscache data structure. But it will be better solution for arbitrary directory tree.
Actually, we know the place where lstat is called heavily, and such lstat is required to return stat result before the file modification of git. So I prefer to use fscache lstat only in such place as I did in #1443 or bit modified like
https://github.com/git-for-windows/git/commit/b7ac8f1882fc2aa7b65a2b27e42f1650cb195ed8

I think fscache with limited scope makes further optimization for other slow lstat easily/safely also, compared to incremental update.

How do you think?

Reply:
I guess while it is heavy-handed, it is still the easiest to reason about...